documentation/client-side-forms/appendix-form-images.asciidoc

```diff
@@ -0,0 +1,136 @@
+---
+title: "Appendix: Image Fields in Client-Side Forms"
+order: 100
+layout: page
+---
+
+ifdef::env-github[:outfilesuffix: .asciidoc]
+
+= Submitting Images in Client-Side Forms
+
+A common requirement is a form field that allows the user to upload an image.
+For instance, an application may let users customize their avatars. This
+tutorial shows how this can be implemented for client-side forms using the
+`vaadin-upload` component for local file selection (or drag and drop), but
+postponing the actual server upload until the form is submitted.
+
+The tutorial app provides editing of a contact card. It  assumes a server-side
+bean where the image is stored as a Base64-encoded string in the `avatarBase64`
+field:
+
+.Contact.java
+[source, java]
+----
+/**
+ * Contact card with base64-encoding image
+ */
+public class Contact {
+    // ...
```

Comment:
Could show how the avatarBase64 field is defined? i.e. in the prototype it has `@Column( length = 10000000 ) // 10 Mb`

Without the annotation, I got an exception
```
Caused by: org.h2.jdbc.JdbcSQLDataException: Value too long for column """AVATAR_BASE64"" VARCHAR(255)"
```

Reply:
I removed the `@Entity` and `@Column` annotations from the example, as from Slack discussion I understood it is considered bad style to use entities as endpoint DTOs. So we may not want to suggest this in our examples, but I am not sure about this?

Reply:
Good point